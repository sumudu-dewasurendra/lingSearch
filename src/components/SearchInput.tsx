import React from 'react';
import {View, TextInput, StyleSheet, Text} from 'react-native';
import {COLORS} from '../theme/colors';

const styles = StyleSheet.create({
  mainWrapper: {
    width: '70%',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    borderWidth: 1,
    borderRadius: 10,
    borderColor: COLORS.BORDER_COLOR,
    backgroundColor: COLORS.SEARCH_INPUT_BACKGROUND_COLOR,
    paddingHorizontal: 10,
  },
  inputWrapper: {
    width: '90%',
    padding: 16,
  },
});

type SearchInputProps = {
  searchText: string;
  onChangeSearchText: (text: string) => void;
  placeholder?: string;
};

const SearchInput = ({
  searchText,
  onChangeSearchText,
  placeholder,
}: SearchInputProps) => {
  return (
    <View style={styles.mainWrapper}>
      <Text>🔍</Text>
      <TextInput
        placeholder={placeholder || 'Search by name'}
        value={searchText || ''}
        onChangeText={text => onChangeSearchText(text)}
        style={styles.inputWrapper}
      />
    </View>
  );
};

export default SearchInput;
