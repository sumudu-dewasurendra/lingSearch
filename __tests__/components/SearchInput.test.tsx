import React from 'react';
import {create} from 'react-test-renderer';
import SearchInput from '../../src/components/SearchInput';
import {expect, test} from '@jest/globals';

test('renders SearchInput component correctly', () => {
  const tree = create(
    <SearchInput
      placeholder="Search"
      onChangeSearchText={(text: string) => console.log(text)}
      searchText=""
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
