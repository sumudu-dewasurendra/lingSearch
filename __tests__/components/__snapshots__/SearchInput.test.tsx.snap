// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchInput component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f9f9f9",
      "borderColor": "#c5c5af",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "width": "70%",
    }
  }
>
  <Text>
    🔍
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search"
    style={
      {
        "padding": 16,
        "width": "90%",
      }
    }
    value=""
  />
</View>
`;
